Test/Uninstall-WhiskeyTool.Tests.ps1

```diff
@@ -47,85 +37,102 @@ function ThenNuGetPackageUninstalled
+    Describe 'Uninstall-WhiskeyTool.when given an NuGet Package with an empty string as Version' {
+        It 'should uninstall all versions of the package' {
+            GivenAnInstalledNuGetPackage 'NUnit.Runners' -WithVersion '2.6.4'
+            GivenAnInstalledNuGetPackage 'NUnit.Runners' -WithVersion '2.6.3'
+            WhenUninstallingNuGetPackage 'NuGet::NUnit.Runners' -WithVersion '' -ErrorAction SilentlyContinue
+            ThenNuGetPackageUnInstalled 'NUnit.Runners' -WithVersion '2.6.4'
+            ThenNuGetPackageUnInstalled 'NUnit.Runners' -WithVersion '2.6.3'
+        }
     }
+
     
+    Describe 'Uninstall-WhiskeyTool.when given an NuGet Package with missing Version' {
+        It 'should uninstall all versions of the package' {
+            GivenAnInstalledNuGetPackage 'NUnit.Runners' -WithVersion '2.6.4'
+            GivenAnInstalledNuGetPackage 'NUnit.Runners' -WithVersion '2.6.3'
+            WhenUninstallingNuGetPackage 'NuGet::NUnit.Runners' -ErrorAction SilentlyContinue
+            ThenNuGetPackageUnInstalled 'NUnit.Runners' -WithVersion '2.6.4'
+            ThenNuGetPackageUnInstalled 'NUnit.Runners' -WithVersion '2.6.3'
+        }
+    }
+
     Describe 'Uninstall-WhiskeyTool.when given an NuGet Package with a wildcard Version' {
-        GivenAnInstalledNuGetPackage -WithVersion '2.*' -ErrorAction SilentlyContinue
-        WhenUninstallingNuGetPackage -WithVersion '2.*' -ErrorAction SilentlyContinue
-        ThenNuGetPackageNotUnInstalled -WithVersion '2.*' -WithError 'Wildcards are not allowed for NuGet packages'
+        It 'should uninstall all versions of the package' {
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '3.6.3'
+            WhenUninstallingNuGetPackage -WithName 'NuGet::NUnit.Runners' -WithVersion '*' -ErrorAction SilentlyContinue
+            ThenNuGetPackageUninstalled -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            ThenNuGetPackageUninstalled -WithName 'NUnit.Runners' -WithVersion '3.6.3'
+        }
+    }
+
+    Describe 'Uninstall-WhiskeyTool.when given an NuGet Package with a pinned wildcard Version' {
+        It 'should uninstall all versions of the package' {
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.6.3'
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '3.6.3'
+            WhenUninstallingNuGetPackage -WithName 'NuGet::NUnit.Runners' -WithVersion '2.*' -ErrorAction SilentlyContinue
+            ThenNuGetPackageUninstalled -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            ThenNuGetPackageUninstalled -WithName 'NUnit.Runners' -WithVersion '2.6.3'
+            ThenNuGetPackageNotUninstalled -WithName 'NUnit.Runners' -WithVersion '3.6.3'
+        }

```

Comment:
Too many tests at this level. `Uninstall-WhiskeyTool` only calls `Uninstall-WhiskeyNuGetPackage` on one line. All you need to verify is that `Uninstall-WhiskeyTool` is passing the correct parameters to `Uninstall-WhiskeyNuGetPackage`. You should only need one test for that. These tests are actually testing the functionality of `Uninstall-WhiskeyNuGetPackage` and should be moved to that function's test fixture if there aren't already equivalent tests there.